### ISSUE TITLE
Submit Workflow: Bookmark Vault

### ISSUE THREAD

**Issue by colomolo** [alfredapp/gallery-edits#64]
### Preflight

- [X] The Alfred team has invited me to submit this workflow.

### The submitted workflow:

- [X] Was authored by me. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#authorship)
- [X] Uses [Workflow Configuration](https://www.alfredapp.com/help/workflows/user-configuration/) or has no user-facing settings. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#configuration)
- [X] Does not have an auto-updater. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#auto-updating)
- [X] Does not contain unsigned and non-notarised binaries. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#signed-binaries)
- [X] Does not download extra software. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#downloading-software)
- [X] Uses keywords with more than one character. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#short-keywords)
- [X] Has a descriptive About section. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#about-section)
- [X] Has an icon of at least 256x256 px. [&#9432;](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Submissions.md#icon-size)

### Homepage

https://github.com/colomolo/alfred-bookmarks

### Gallery page content

![Bookmark Vault](https://github.com/user-attachments/assets/2591d2e7-b7e0-4e18-80b3-ab891579d31e)

## Usage
Manage your links, quick search notes or checklists in Markdown files.

All your bookmarks are stored locally on your computer and independent from any app.

After installing the workflow you should specify the working folder in workflow configuration. Bookmark Vault will scan this folder and treat each .md file as a list. If for some reason you will delete Bookmarks or even Alfred, your once created lists will still stay with you.

* `bk` to see your lists:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/5adf9154-3b2a-442f-b1ab-c5974f39de40">

* `bkadd` to create and add new item to one of your lists:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/2979bc99-e6bf-4b04-a4db-d7424336dcc9">

* `bktab` (or we recommend to assign Hotkey for this) to add the current active tab of the frontmost browser to the list:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/8f626c48-72dc-4d8f-9127-9c1642e3f199">

### List actions
* <kbd>↩︎</kbd> Open list
* <kbd>⌘</kbd><kbd>↩︎</kbd> Open list file in text editor
* <kbd>⇧</kbd><kbd>↩︎</kbd> Show checked items
* <kbd>⌘</kbd><kbd>⌥</kbd><kbd>⇧</kbd><kbd>↩︎</kbd> Delete list permanently

### List items 
There are two list item types – URLs and search queries. URLs are opened in the browser and plain text is used in web search.

You also can add checkboxes to list items. This is handy when you want to open your bookmark only once. For example, you can use it for a list of articles that you need to go through. Checkbox items will be checked on opening and hidden from the list. If you still want to see them, hold <kbd>⇧</kbd> while picking a list.

<img width="764" alt="image" src="https://github.com/user-attachments/assets/ae03d265-bff0-4495-8d7a-b7163f1641f4">

* <kbd>↩︎</kbd> Open list item
* <kbd>⇧</kbd><kbd>↩︎</kbd> Convert item to checkbox/plain item
* <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↩︎</kbd> Check/uncheck checkbox item
* <kbd>⌘</kbd><kbd>⌥</kbd><kbd>⇧</kbd><kbd>↩︎</kbd> Delete item permanently

Bookmark Vault will try to present URLs in a convenient way. If it's possible to get page's title (by fetching the page), it will be shown as an item title. You can also specify it manually by formatting link in Markdown format. For example adding `[Transport Museum](https://www.ltmuseum.co.uk/)` as an argument to `bkadd` command will later be represented with a title and URL as a subtitle below.

Happy bookmarking!

### Tags

list, bookmark, checklist

**Comment by vitorgalvao** [comment]
Thank you, but to be accepted the submission must follow the requirements. The provided instructions don’t follow the consistent [Style Guide of Gallery pages](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Workflow%20Page%20Style%20Guide.md), and neither do [the screenshots](https://github.com/alfredapp/gallery-edits/blob/main/Documentation/Taking%20Good%20Screenshots.md). All of these are linked to and documented in the submission form.

**Comment by colomolo** [comment]
> Thank you, but to be accepted the submission must follow the requirements. 

Thank you, I will address these.

**Comment by colomolo** [comment]
@vitorgalvao Done.

**Comment by vitorgalvao** [comment]
Better, but the screenshots are still incorrect. They must have no background. The linked instructions go into detail about that. Also, the top one has the text cursor in it.

And actions should be described in order with respective screenshots. Example:

```
Add the current browser tab to a list via the `bktab` keyword.

SCREENSHOT

Do SOMETHING via the `bkadd` keyword.

SCREENSHOTS

See your lists with the `bk` keyword

SCREENSHOT
```

It’s not clear from the current instructions what `bkadd`  does.

**Comment by colomolo** [comment]
@vitorgalvao I've updated screenshots and description again. Let me know what else I can improve.

**Comment by vitorgalvao** [comment]
It still needs a few tweaks but I can take it from there.

The one thing missing from your side is that you have not published the workflow in the assets of [the release](https://github.com/colomolo/alfred-bookmarks/releases/tag/v1.3.0). Can’t proceed without that.

**Comment by vitorgalvao** [comment]
@colomolo It still needs tweaks but I can take it from there.

But you have not published the workflow in the assets of [the release](https://github.com/colomolo/alfred-bookmarks/releases/tag/v1.3.0). Can’t proceed without that.

**Comment by colomolo** [comment]
@vitorgalvao oh, that was a bug after renaming the workflow. Fixed now, here's 1.3.1 release: https://github.com/colomolo/alfred-bookmarks/releases/tag/v1.3.1

> Also, and this could be for the next version, the first experience could be improved. If there’s no list in the folder when using bkadd, it just fails. It should inform the user of what’s wrong and possibly offer a next step. Same thing if the folder does not exist.

Great catch, will fix that!
